### PR TITLE
Feature/1714/328 jwtutils class name

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Rename `VerifiableCredentialsFactory` to `JwtUtils` and move it to identity-did-service (single place where it's used)

## Why it does that

The class had the doc `Convenience/helper class to generate, verify and deserialize verifiable credentials, which are, in fact, Signed JSON Web Tokens (JWTs).`

In fact the implementation is only based on DID and JWT, and does not contain any claims from the verifiable credentials data model. It is used for generating and validating short-lived authentication tokens in synchronous HTTP requests - and not for managing verifiable credentials, which will be done in other components (IdentityHub, MVD).

Trying to extend the class to also support the verifiable credential data model wouldn't seem to be a good idea, since there are specifics to usage as a web request token (audience, expiration, etc.)

## Linked Issue(s)

#328 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/JwtUtils.java [hunk 1]
```diff
@@ -36,31 +36,28 @@
...
+     * @param issuerAndSubject the value of the token issuer and subject claims. For Distributed Identity, these values are identical.
+     * @param audience         the value of the token audience claim, e.g. the IDS Webhook address.
+     * @param clock            clock used to get current time
      * @return a {@code SignedJWT} that is signed with the private key and contains all claims listed
      */
-    public static SignedJWT create(PrivateKeyWrapper privateKey, String issuer, String audience, Clock clock) {
-        var claimsSet = new JWTClaimsSet.Builder().issuer(issuer)
-                .issuer(issuer)
-                .subject(VERIFIABLE_CREDENTIAL)
+    static SignedJWT create(PrivateKeyWrapper privateKey, String issuerAndSubject, String audience, Clock clock) {
```
Comment (cpeeyush) [inline extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/JwtUtils.java:57]: nit: issuerAndSubject as per javadoc seems like here we always same value for issue and subject for now. But it looks little strange if we use one input argument for two different purposes. 

may we have different args here one for issue and one for subject and can have an overloaded method which have `String issuerAndSubject` for this use case.

But this more of my coding style choice so if you do not prefer it then we leave it. Thanks.

File: extensions/iam/decentralized-identity/identity-did-service/build.gradle.kts [hunk 2]
```diff
@@ -3,12 +3,15 @@ plugins {
...
 
+val datafaker: String by project
+
 dependencies {
     api(project(":extensions:iam:decentralized-identity:identity-did-spi"))
     implementation(project(":extensions:iam:decentralized-identity:identity-did-crypto"))
 
     testImplementation(testFixtures(project(":extensions:iam:decentralized-identity:identity-common-test")))
     testImplementation(project(":extensions:junit"))
+    testImplementation("net.datafaker:datafaker:${datafaker}")
```
Comment (cpeeyush) [inline extensions/iam/decentralized-identity/identity-did-service/build.gradle.kts:14]: nit: seems unused ?

**Review by cpeeyush — APPROVED** [review]
Approved with suggestion.

File: extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/JwtUtils.java (hunk 1 above)
Comment (algattik) [reply extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/JwtUtils.java]: Thanks, implemented along those lines

**Review by chrislomonico — APPROVED** [review]

**Pull request merged**